tests/ssl-opt.sh

```diff
@@ -2375,6 +2375,17 @@ run_test    "Extended Master Secret: client enabled, server SSLv3" \
             -C "session hash for extended master secret" \
             -S "session hash for extended master secret"
 
+run_test    "Extended Master Secret: both enabled, both enforcing, DTLS" \
```

Comment:
I was going to comment that we should also have tests with one party not disabling it so that we can make sure that enforcing actually works... and then I read the commit message :) Perhaps in the distant future we can extend the test framework to be able to test a client with a config against a server with another config, but that's a long-term project far out of scope of this PR, so you did all we could do now.